### PR TITLE
Fix net5.0 Windows TFMs

### PR DESCRIPTION
**Pull request by tmat**
Fix warnings reported by newer .NET SDK.

### CODE REVIEW

File: src/Workspaces/CoreTestUtilities/Remote/InProcRemostHostClient.cs [hunk 1]
```diff
@@ -384,6 +384,9 @@ public override int WriteTimeout
 
                 public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken) => _stream.CopyToAsync(destination, bufferSize, cancellationToken);
 
+#if NET5_0
+                [Obsolete]
```
Comment (dibarbet) [inline src/Workspaces/CoreTestUtilities/Remote/InProcRemostHostClient.cs:388]: I saw the tfm warnings, but what issue is this fixing?

Comment (tmat) [reply src/Workspaces/CoreTestUtilities/Remote/InProcRemostHostClient.cs:388]: The warning not being reported.

Comment (tmat) [reply src/Workspaces/CoreTestUtilities/Remote/InProcRemostHostClient.cs:388]: Seems like the newer compiler is reporting more warnings than the previous version.

Comment (tmat) [reply src/Workspaces/CoreTestUtilities/Remote/InProcRemostHostClient.cs:388]: Odd, now I see 
```
error CS0809: Obsolete member 'InProcRemoteHostClient.InProcRemoteServices.WrappedStream.InitializeLifetimeService()' overrides non-obsolete member 'MarshalByRefObject.InitializeLifetimeService()
```

Comment (tmat) [reply src/Workspaces/CoreTestUtilities/Remote/InProcRemostHostClient.cs]: Turn out we can remove this override.

**Review by davidwengier — APPROVED** [review]
~Should the SDK change too, from `Microsoft.NET.Sdk.WindowsDesktop` to `Microsoft.NET.Sdk` at the same time?~ Don't do this, I'm wrong.

**Comment by tmat** [comment]
@davidwengier IDK, should it?

**Comment by davidwengier** [comment]
Looks like, no, it shouldn't! :)

**Pull request merged**